File: ExampleCodes/Java/src/Logarithmic.java
public class Logarithmic {

        //lets introduce a for loop, but instead of
        //just running through the loop
        //one by one all the way to the end
        //we will double the value of i each time
        //meaning we will reach the end of the loop sooner
        //this is the second lowest complexity class behind
        //constant time
        public static void main(String[] args) {

                System.out.println("Starting...");

                final long startTime = System.currentTimeMillis();
                int n = 16;
                for (int i = 1; i < n; i = i*2) {
                        System.out.println("i is now " + i);
                }
                final long endTime = System.currentTimeMillis();

                System.out.println("Took " + (endTime - startTime) 
                                + " milliseconds to complete");
                System.out.println("This is an example of O(log 16) with 4 iterations as shown, note that this is O(log n) and int n here is 16");
        }



}

File: ExampleCodes/Javascript/Logarithmic.js
console.log("Starting...");

const Logarithmic = (n) => {
  //O(log n) complexity
  for (let i = 1; i <= n; i = i * 2) {
    console.log("i is " + i);
  }
};

const { performance } = require("perf_hooks");
const time1 = performance.now();
Logarithmic(16); //this is n
const time2 = performance.now();
console.log(`Elapsed time: ${(time2 - time1) / 1000} seconds`);
